### ISSUE TITLE
Add regex selector to `nodeRules` and `childNodeRules`

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#227]
## Spec

Currently:

```json
{
  "nodeRules": [
    {
      "selector": "[class^='c-']",
      "rules": { ... }
    }
  ]
}
```

New feature:

```jsonc
{
  "nodeRules": [
    {
      "regexSelector": {
        "nodeName": "/^x-/i",
        "attrName": "/^data-(.+)/i", // Capturable
        "attrValue": "/^https?/i"
      },
      "rules": {
        "class-naming": "/^c-{{$1}}/" // Use captured group with mustache format
      }
    }
  ]
}
```

---

## Use case

Checking to rule that class-naming of structures BEM-like.

```jsonc
{
  "rules": {
    "class-naming": "/.+/"
  },
  "childNodeRules": [
    {
      "regexSelector": {
        "attrName": "class",
        "attrValue": "/^(?<ComponentName>[A-Z][a-z0-9]+)$/"
      },
      "rules": {
        "class-naming": {
          "value": ["/^{{ComponentName}}__[a-z0-9]+$/", "/^([A-Z][a-z0-9]+)$/"],
          "reason": "Do not allow include the element in a no-own block."
        }
      }
    }
  ]
}
```

👍 Correct:

```html
<section class="Card">
  <div class="Card__header">
    <div class="Heading"><h3 class="Heading__lv3">Title</h3></div>
  </div>
  <div class="Card__body">
    <div class="List">
      <ul class="List__group">
        <li>...</li>
        <li>...</li>
        <li>...</li>
      </ul>
    </div>
  </div>
</section>
```

👎 Incorrect:

```html
<section class="Card">
  <div class="Card__header">
    <!-- 👎 Element in Element belong Another Block -->
    <h3 class="Heading__lv3">Title</h3>
  </div>
  <div class="Card__body">
    <!-- 👎 Element in Element belong Another Block -->
    <ul class="List__group">
      <li>...</li>
      <li>...</li>
      <li>...</li>
    </ul>
  </div>
</section>
```
